plexapi/library.py

```diff
@@ -549,15 +582,32 @@ def search(self, title=None, sort=None, maxresults=999999, libtype=None, **kwarg
             args['sort'] = self._cleanSearchSort(sort)
         if libtype is not None:
             args['type'] = utils.searchType(libtype)
-        # iterate over the results
-        results, subresults = [], '_init'
-        args['X-Plex-Container-Start'] = 0
-        args['X-Plex-Container-Size'] = min(X_PLEX_CONTAINER_SIZE, maxresults)
-        while subresults and maxresults > len(results):
+
+        results = []
+        subresults = []
+
+        if maxresults is not None:
+            container_size = min(container_size, maxresults)
+        while True:
             key = '/library/sections/%s/all%s' % (self.key, utils.joinArgs(args))
-            subresults = self.fetchItems(key)
-            results += subresults[:maxresults - len(results)]
-            args['X-Plex-Container-Start'] += args['X-Plex-Container-Size']
+            subresults = self.fetchItems(key, container_start=container_start,
+                                         container_size=container_size)
+            results.extend(subresults)
+
+            # self.totalSize is not used as a condition in the while loop as
+            # this require a additional http request.
+            # self.totalSize is updated from .fetchItems
+            if maxresults is not None:
+                res = min(maxresults, self.totalSize)
+                container_size = min(container_size, maxresults - len(results))
+            else:
+                res = self.totalSize
+
+            if res <= len(results):
```

Comment:
I think this condition is missing special handling for a request with an initial `container_start > 0` because then the total result set is smaller than `self.totalSize`. So if your library has 1000 items and you start with `container_start = 100` you only expect to retrieve 900 items.

This might actually result in an endless loop because you increase the `container_start` which will probably result in Plex not returning any items if `container_start > totalSize` so `results` will not grow anymore and this condition will never be valid.

Reply:
Yeah, let just break the loop if the subresult is empty. 👍